explorer/interpreter/resolve_control_flow.cpp

```diff
@@ -53,8 +53,11 @@ static auto ResolveControlFlow(Nonnull<Statement*> statement,
         }
         (*function)->saw_return_in_auto = true;
       }
-      auto& ret = cast<Return>(*statement);
-      ret.set_function((*function)->declaration);
+      if ((*function)->declaration->kind() ==
+          DeclarationKind::FunctionDeclaration) {
```

Comment:
Should this be `CallableDeclaration`?

Reply:
No, in this case I need explicit a FunctionDeclaration for Return. I reluctantly wanted to customize the ReturnStatement to handle CallableDeclaration

Reply:
I'm worried that this means `return` statements in destructors won't work. Am I missing something?

Reply:
Done!